src/controllers/utils/minimizeWords.ts

```diff
@@ -49,13 +67,14 @@ const minimizeWords = (words, version) => {
     }
 
     if (version === Version.VERSION_2 && minimizedWord.dialects?.length) {
-      minimizedWord.dialects = minimizedWord.dialects?.map((dialect) => {
-        let minimizedDialect = omit(dialect, ['variations', 'id', '_id']);
-        if (!minimizedDialect.pronunciation) {
-          minimizedDialect = omit(minimizedDialect, ['pronunciation']);
-        }
-        return minimizedDialect;
-      });
+      if (Array.isArray(minimizedWord.dialects))
+        minimizedWord.dialects = minimizedWord.dialects?.map((dialect) => {
```

Comment:
remove the `?.` if you have already checked to see if `minimizedWord.dialects` is an array